internal/bminventory/inventory.go

```diff
@@ -2289,6 +2296,15 @@ func (b *bareMetalInventory) RegisterHost(ctx context.Context, params installer.
 		return installer.NewRegisterHostInternalServerError().
 			WithPayload(common.GenerateError(http.StatusInternalServerError, err))
 	}
+
+	if cluster.KubeKeyNamespace != "" {
```

Comment:
Maybe it should be inside `CreateAgentCR`? it gets the namespace anyway right?

Reply:
Done